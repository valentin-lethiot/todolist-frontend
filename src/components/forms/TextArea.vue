<script setup lang="ts">
type Props = {
  label: string
  modelValue: string
}
defineProps<Props>()

const emit = defineEmits(['update:modelValue'])

const emitNewValue = (event: any) => {
  emit('update:modelValue', event.target.value)
}
</script>

<template>
  <div>
    <label class="block mb-1 text-md font-medium text-white">{{ label }}</label>
    <textarea
      :value="modelValue"
      rows="10"
      class="bg-gray-50 border text-sm rounded-lg block w-full p-2 bg-gray-700 border-gray-600 text-white"
      required
      @input="emitNewValue"
    />
  </div>
</template>

<style scoped>
textarea {
  resize: none;
}
</style>
