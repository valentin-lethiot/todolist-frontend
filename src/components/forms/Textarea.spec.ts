import { describe, beforeEach, expect, test } from 'vitest'
import TextArea from './TextArea.vue'
import { mount, VueWrapper } from '@vue/test-utils'

describe('TextArea', () => {
  let wrapper: VueWrapper

  const defaultProps = {
    label: 'label',
    modelValue: '',
  }

  beforeEach(() => {
    wrapper = mount(TextArea, {
      props: defaultProps,
    })
  })

  describe('render', () => {
    test('should render the textarea', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('events', () => {
    test('should emit update:modelValue event when textarea changes', async () => {
      const textarea = wrapper.find('textarea')
      await textarea.setValue('new value')

      expect(wrapper.emitted('update:modelValue')).toBeTruthy()
      expect(wrapper.emitted('update:modelValue')![0]).toEqual(['new value'])
    })
  })

  describe('props', () => {
    test('should display the label', () => {
      const label = wrapper.find('label')
      expect(label.text()).toBe('label')
    })

    test('should bind the modelValue to textarea value', () => {
      const textarea = wrapper.find('textarea')
      expect(textarea.element.value).toBe('')

      wrapper = mount(TextArea, {
        props: {
          ...defaultProps,
          modelValue: 'test value',
        },
      })

      expect(wrapper.find('textarea').element.value).toBe('test value')
    })
  })
})
