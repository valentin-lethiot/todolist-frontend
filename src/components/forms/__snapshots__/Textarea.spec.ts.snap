// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TextArea > render > should render the textarea 1`] = `"<div data-v-993744c5=""><label data-v-993744c5="" class="block mb-1 text-md font-medium text-white">label</label><textarea data-v-993744c5="" rows="10" class="bg-gray-50 border text-sm rounded-lg block w-full p-2 bg-gray-700 border-gray-600 text-white" required="" value=""></textarea></div>"`;
